server/src/test/java/org/opensearch/gateway/AsyncShardFetchTests.java

```diff
@@ -100,14 +125,26 @@ public void testClose() throws Exception {
         AsyncShardFetch.FetchResult<Response> fetchData = test.fetchData(nodes, emptyMap());
         assertThat(fetchData.hasData(), equalTo(false));
         assertThat(test.reroute.get(), equalTo(0));
+        // counter remains 0 because fetch is ongoing
+        verify(asyncFetchSuccessCounter, times(0)).add(anyDouble());
+        verify(asyncFetchFailureCounter, times(0)).add(anyDouble());
 
         // fire a response, wait on reroute incrementing
         test.fireSimulationAndWait(node1.getId());
+        // counter goes up because fetch completed
+        verify(asyncFetchSuccessCounter, times(1)).add(1.0);
+        verify(asyncFetchSuccessCounter, times(1)).add(anyDouble());
```

Comment:
why is this assertion needed? basically double verify on success/ failure metrics at times

Reply:
first assertion verifies that counter was called with argument 1.0 once.
second assertion verifies that the counter was called exactly once. (for any argument)

this is a stronger assertion and verifies that async fetch happened exactly once, and the argument was 1.0. 


this is a stronger assertion as we are validating

Reply:
then would using atMost be better here?